core/src/main/kotlin/xtdb/vector/IVectorIndirection.kt

```diff
@@ -0,0 +1,37 @@
+package xtdb.vector
+
+interface IVectorIndirection {
+    fun valueCount(): Int
+
+    fun getIndex(idx: Int): Int
+
+    operator fun get(idx: Int): Int {
+        return getIndex(idx)
+    }
```

Comment:
in Kotlin, this is usually one-lined: 
```suggestion
    operator fun get(idx: Int) = getIndex(idx)
```